swagger.yaml

```diff
@@ -32,6 +39,24 @@ paths:
         500:
           $ref: '#/responses/InternalError'
   /recipes/{recipe_id}:
+    post:
+      summary: "Create a recipe"
+      description: |
+        Create a recipe and add to mongo
+      parameters:
+      - $ref: '#/parameters/recipe_id' 
+      - $ref: '#/parameters/new_recipe'
```

Comment:
new-recipe missing definition of fields to be handled in request, think about whether these fields are optional or mandatory. If you don't know then speak to Jim he will know what the json should look like, what will and will not work.

Currently I can create a recipe with an empty object - so once the fields above have been defined and ocumented in swagger spec then we can look at applying some validation in the code.

Reply:
The definition of the fields of `new_recipe` is mentioned in the `schema` which is mentioned in `parameters -> new_recipes`. If I define the fields, would I not be repeating the same information?

Reply:
While speaking to Jim, all these fields are mandatory and therefore, as you said, validation needs to be put in place. We shouldn't create a recipe with an empty object because it is pointless and not in practice

Reply:
You can use inheritance in schema so there is less duplication; but we should be defining what the user has to send to us currently we are saying send us:

```
{
  "input_files": [
    {}
  ]
}
```

We should be defining what fields they should send us and what structure it should take. The input structure maybe different to the output structure (and sometimes the stored data will take on a different structure)

Reply:
Resolved in commit _"Update swagger to remove readOnly for PUT and POST parameters and add required properties"_

Reply:
Do you also need to consider showing the user that auth is required for the post endpoints? Create a security definition: 
example:
```
securityDefinitions:
  ServiceToken:
    type: apiKey
    in: header
    name: service-token
```

then use it in the post path:

```
    post:
      security:
        - ServiceToken: []
```

This is what I use in the sessions api, not sure if you need something similar here. Worth checking.

Reply:
Done in commit "update swagger spec with auth details"